go_lib/dependency/cr/cr.go

```diff
@@ -112,12 +112,12 @@ func (r *client) Digest(tag string) (string, error) {
 	return d.String(), nil
 }
 
-func readAuthConfig(configPath string) (authn.AuthConfig, error) {
-	dockerConfigBytes, err := os.ReadFile(configPath)
+func readAuthConfig(dockerCfgBase64 string) (authn.AuthConfig, error) {
+	dockerCfg, err := base64.StdEncoding.DecodeString(dockerCfgBase64)
```

Comment:
````
u1, _ := url.Parse(repo)
u1.Host
```

key := u1.Host

  var tmp map[string]interface{}
  json.Unmarshal(a, &tmp)

for _, a := range tmp {
  u2, _ := url.Parse(a)
  if u2 != u1 { 
    continue
  }
...
}